pkgs/servers/nextcloud/packages/apps/recognize.nix

```diff
@@ -0,0 +1,103 @@
+{
+  stdenv,
+  fetchurl,
+  lib,
+  nodejs,
+  python311,
+  util-linux,
+  ffmpeg,
+}:
+stdenv.mkDerivation rec {
+  pname = "nextcloud-app-recognise";
+  version = "8.1.1";
+
+  srcs = [
+    (fetchurl {
+      inherit version;
+      url = "https://github.com/nextcloud/recognize/releases/download/v${version}/recognize-${version}.tar.gz";
+      hash = "sha256-RIa2HzX2hfgS7KHGGBsxEdUCqDqXeAs3Xi288qA1gU4=";
+    })
+
+    (fetchurl {
+      inherit version;
+      url = "https://github.com/nextcloud/recognize/archive/refs/tags/v${version}.tar.gz";
+      hash = "sha256-op1Fdm40d8V6V+rfne0ECr60xRjBYrBPrCD3kaNeFfY=";
+    })
+    (fetchurl {
+      inherit version;
+      url = "https://storage.googleapis.com/tensorflow/libtensorflow/libtensorflow-cpu-linux-x86_64-2.9.1.tar.gz";
+      hash = "sha256-f1ENJUbj214QsdEZRjaJAD1YeEKJKtPJW8pRz4KCAXM=";
+    })
+
+  ];
+
+  unpackPhase = ''
+    # Merge the app and the models from github
+    tar -xzpf "${builtins.elemAt srcs 0}" recognize;
+    tar -xzpf "${builtins.elemAt srcs 1}" recognize-${version}/models;
+    mv recognize-${version}/models recognize
+
+    # Place the tensorflow lib at the right place for building
+    cd recognize/node_modules/@tensorflow/tfjs-node/deps
+    tar -xzpf "${builtins.elemAt srcs 2}"
+    cd -
+  '';
+
+  patchPhase = ''
+    #hardcode node
+    sed -i "/'node_binary'/s:'""':'Nix Controled':" recognize/lib/Service/SettingsService.php
+
+    substituteInPlace recognize/lib/**/*.php \
+      --replace-quiet "\$this->settingsService->getSetting('node_binary')" "'${nodejs}/bin/node'"
```

Comment:
`--replace-fail` is more appropriate here so we notice on updates that we need to update the patching, no?

Reply:
I just looked at the source of substitute in place and since most of the files do not read the settings, I think it would fail. (While reading I discovered that I could combine the multiple invocations into one)

I added a check to validate that we indeed removed all occurrences

Reply:
Oh right, there's a glob. OK yes fine.

> I added a check to validate that we indeed removed all occurrences

Di you push it or am I just not seeing it?

Reply:
I just pushed it now https://github.com/NixOS/nixpkgs/pull/333545/commits/69b961b12733745f6bbebc02c8cbae293b479e6b